### ISSUE TITLE
SVG.invent does specifiy SVG.Container as default parent

### ISSUE THREAD

**Issue by Fuzzyma** [svgdotjs/svg.js#611]
This is maybe related to #610.
All the methods to create a shape are added to SVG.Container by default.
However there are parent elements like `SVG.G` which shouldnt be container cause they cant have a viewbox. Group should be an SVG.Parent instead. That said it would make sense to add all the contructor methods to SVG.Parent instead of SVG.Container by default.
Same goes for gradient as seen in #610.

Reworking this will most likely not break any code because SVG.Container only adds the viewbox - nothing more. So there shouldnt be a difference

**Comment by Fuzzyma** [comment]
After thinking about that: Moving the elements consructor to the SVG.Parent class brings other problems.
E.g. a user shouldnt be able to create a rect in a gradient or something like that. However: Having the viewport function on all containers (e.g. group) is also wrong. Maybe we need to subclass here again

**Comment by wout** [comment]
Yes, I have been thinking that as well a while ago. Something like `SVG.Frame` or `SVG.Framable` (as in, it's possible to frame the content differently that it actually is).

**Comment by Fuzzyma** [comment]
So we have to differentiate between:
- Elements which can have children (like svg, defs, gradient, textPath and all others)
- Elements which can create shapes (group, svg, nested, symbol...)
- Elements which also can have a viewport (svg, nested, symbol - no more i know)

**Comment by wout** [comment]
Yeah, that's what I mean. But I don't know if we will get there with a sub-class alone.

```js
SVG.Element > SVG.Parent > SVG.Container > SVG.Frame > SVG.Doc
```

Where `SVG.Frame` carries the `viewbox()` method.

Now that I think of it, maybe it's just better to extend the classes required classes with the viewbox method:

```js
SVG.extend(SVG.Doc, SVG.Nested, SVG.Symbol, {
  ...
})
```

**Comment by Fuzzyma** [comment]
Yes that sounds even better

**Comment by Fuzzyma** [comment]
All this elements can have a viewbox:
- SVG
- symbol
- image
- foreignObject
- marker
- pattern
- view

**Comment by Fuzzyma** [comment]
This was solved as part of #634 ([3.0 branch](https://github.com/svgdotjs/svg.js/tree/3.0.0)). `SVG.Container` is the valid default for constructor methods. The `viewbox` method is only available on the above elements now.
